pkg/controllers/scheduling/placeholder_controller.go

```diff
@@ -0,0 +1,231 @@
+package scheduling
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	cache "k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	clusterclient "github.com/open-cluster-management/api/client/cluster/clientset/versioned"
+	clusterinformerv1alpha1 "github.com/open-cluster-management/api/client/cluster/informers/externalversions/cluster/v1alpha1"
+	clusterlisterv1 "github.com/open-cluster-management/api/client/cluster/listers/cluster/v1"
+	clusterlisterv1alpha1 "github.com/open-cluster-management/api/client/cluster/listers/cluster/v1alpha1"
+	clusterapiv1alpha1 "github.com/open-cluster-management/api/cluster/v1alpha1"
+)
+
+// decisionPlaceholderController creates empty decisions as placeholders in status of PlacementDecisions.
+// The contoller support placements with Spec.NumberOfCluster specified only.
+// TODO: Add support to placements without Spec.NumberOfCluster specified by enhanceing this controller
+// or creating a separate one.
+type decisionPlaceholderController struct {
+	clusterClient           clusterclient.Interface
+	clusterLister           clusterlisterv1.ManagedClusterLister
+	placementLister         clusterlisterv1alpha1.PlacementLister
+	placementDecisionLister clusterlisterv1alpha1.PlacementDecisionLister
+}
+
+// NewDecisionPlaceholderController return an instance of decisionPlaceholderController
+func NewDecisionPlaceholderController(
```

Comment:
I missed where in this controller we build the list of clusters.

Reply:
This controller does not list clusters. In func updateClusterDecisions(), it creates empty cluster decisions as placeholders in the status of placementdecisions, and guarantees each placementdecision contains the correct number of cluster decisions (some of them might be empty). Another controller (created in PR https://github.com/open-cluster-management/placement/pull/7) will monitor placementdecisions and list clusters to pick up feasible ones for the unscheduled cluster decisions (the empty ones).